### PR TITLE
Adds python 3.11 testing support

### PR DESCRIPTION
**Pull request by kingbuzzman**
Adds python 3.11 testing support

### CODE REVIEW

**Comment by kingbuzzman** [comment]
Waiting on https://github.com/actions/runner-images/issues/6459 to finish

**Comment by kingbuzzman** [comment]
@francoisfreitag looks like the demo flask app is broken. They release 2.x.x recently, i'm guessing they broke something.

**Comment by kingbuzzman** [comment]
@francoisfreitag this should be good to go.

**Comment by francoisfreitag** [comment]
Thanks for the bump! I’m failing to reproduce the warning from https://github.com/FactoryBoy/factory_boy/pull/982/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R65 on Python 3.11 on my machine. Do you remember what the warning was about, and where it’s originating from?

**Comment by kingbuzzman** [comment]
It's thrown on [py311-django32-mongo-alchemy-sqlite](https://github.com/FactoryBoy/factory_boy/actions/runs/3337692146/jobs/5524360244) (and others..)

**Comment by kingbuzzman** [comment]
@francoisfreitag It's thrown on [py311-django32-mongo-alchemy-sqlite](https://github.com/FactoryBoy/factory_boy/actions/runs/3337692146/jobs/5524360244) (and others..)

**Comment by francoisfreitag** [comment]
/home/runner/work/factory_boy/factory_boy/.tox/py311-django32-mongo-alchemy-sqlite/lib/python3.11/site-packages/django/http/request.py

Great, it has been fixed upstream. https://github.com/django/django/commit/34e2148fc725e7200050f74130d7523e3cd8507a

**Comment by francoisfreitag** [comment]
Can drop commit “[So cgi can throw its warning](https://github.com/FactoryBoy/factory_boy/pull/982/commits/76918c0f93aa619789047036eb122c0300087695)”

File: Makefile [hunk 1]
```diff
@@ -62,6 +62,7 @@ test:
 		-Wdefault:"Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working":DeprecationWarning:: \
 		-Wdefault:"set_output_charset() is deprecated":DeprecationWarning:: \
 		-Wdefault:"parameter codeset is deprecated":DeprecationWarning:: \
+		-Wdefault:"'cgi' is deprecated and slated for removal in Python 3.13":DeprecationWarning:: \
```
Comment (francoisfreitag) [inline Makefile:65]: Can you add a TODO to remove the CGI warning in when dropping support for Django 4.1 ?

**Review by francoisfreitag — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (kingbuzzman) [reply Makefile]: Done.

**Comment by kingbuzzman** [comment]
@francoisfreitag All is ready now.

**Pull request merged**